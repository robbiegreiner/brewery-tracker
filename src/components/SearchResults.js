import React from 'react';
import PropTypes from 'prop-types';
import BeerCard from './BeerCard.js';
import BreweryCard from './BreweryCard.js';
import { Link } from 'react-router-dom';
import CityCard from './CityCard.js';

const SearchResults = ({ searchResults, getBrewery, setCurrentBeer, searchType, user, addFavoriteBrewery, addFavoriteBeer, favorites, errorMessage }) => {

  const beerResults = searchResults.filter( result => {
    return result.style;
  });

  const breweryResults = searchResults.filter( result => {
    return result.type === "brewery";
  });

  const cityResults = searchResults.filter( result => {
    return result.breweryId;
  });

  if (favorites){
    const favoriteIDs = Object.keys(favorites).map( favorite => {
      return favorites[favorite].id;
    });

    breweryResults.forEach( brewery => {
      if (favoriteIDs.includes(brewery.id)){
        brewery.isFav = true;
      }
    });

    beerResults.forEach( beer => {
      if (favoriteIDs.includes(beer.id)){
        beer.isFav = true;
      }
    });
  }



  const breweryCards = breweryResults.map( brewery => {
    return <BreweryCard
      user={user}
      addFavoriteBrewery={addFavoriteBrewery}
      favorites={favorites}
      getBrewery={getBrewery}
      brewery={brewery}
      key={brewery.id}/>;
  });


  const beerCards = beerResults.map( beer => {
    return <BeerCard
      user={user}
      favorites={favorites}
      addFavoriteBeer={addFavoriteBeer}
      setCurrentBeer={setCurrentBeer}
      beer={beer}
      key={beer.id}/>;
  });

  const cityCards = cityResults.map( brewery => {
    return <CityCard
      getBrewery={getBrewery}
      favorites={favorites}
      addFavoriteBrewery={addFavoriteBrewery}
      user={user}
      brewery={brewery}
      key={brewery.id}/>;
  });

  const showError = () => {
    if (errorMessage || !searchResults.length) {
      return (
        <div>
          <h2>No results found, please enter a valid brewery, beer or city.</h2>
          <Link to='/'><h2 className='search-again'>Click Here To Search Again!</h2></Link>
        </div>
      );
    }
  };

  return (
    <div className='search-results'>
      <h6
        className='back-button'
        onClick={() => window.history.back()}><u>Go Back</u></h6>
      <h1>Search Results</h1>
      <div className='card-container'>
        { showError() }
        { searchType === 'beer' ? beerCards : null}
        { searchType === 'brewery' ? breweryCards : null}
        { searchType === 'city' ? cityCards : null}
      </div>
    </div>
  );
};


SearchResults.propTypes = {
  searchResults: PropTypes.array,
  getBrewery: PropTypes.func,
  setCurrentBeer: PropTypes.func,
  searchType: PropTypes.string,
  user: PropTypes.object,
  addFavoriteBeer: PropTypes.func,
  addFavoriteBrewery: PropTypes.func,
  favorites: PropTypes.object,
  errorMessage: PropTypes.string
};

export default SearchResults;
