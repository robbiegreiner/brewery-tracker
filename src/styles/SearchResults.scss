.search-results {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 100px;
}

.search-container {
  display: flex;
}

.search-container button {
    margin: 20px 0px 10px 0px;
    background-color: #868e96;
    border-color: #868e96;
    color: white;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    border: 2px solid #41403D;
    padding: .75rem 1rem;
    font-size: 16px;
    letter-spacing: 2px;
    line-height: 1.25;
    border-radius: .25rem;
    transition: all .15s ease-in-out;
}

.search-container button:hover {
  background-color: #364248;
}

.search-container input {
  height: 42px;
  margin-top: 20px;
  border-radius: .25rem;
  font-size: 16px;
  border: 2px solid #41403D;
}

.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  width: 95%;
}
