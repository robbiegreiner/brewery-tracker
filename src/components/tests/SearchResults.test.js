import React from "react";
import SearchResults from '../SearchResults';
import { shallow } from "enzyme";

describe('SearchResults component', () => {
  const mockFn = jest.fn();

  const wrapper = shallow(<SearchResults
    user={{}}
    setCurrentBeer={mockFn}
    getBrewery={mockFn}
    searchType={'beer'}
    searchResults={[{}, {}]}/>);


  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});
