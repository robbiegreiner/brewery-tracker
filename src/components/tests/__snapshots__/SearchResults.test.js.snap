// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    getBrewery={[Function]}
    searchResults={
        Array [
            Object {},
            Object {},
          ]
    }
    searchType="beer"
    setCurrentBeer={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Search Results
</h1>,
        <div
          className="card-container"
>
          
</div>,
      ],
      "className": "search-results",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Search Results",
        },
        "ref": null,
        "rendered": "Search Results",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            Array [],
            null,
            null,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          null,
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Search Results
</h1>,
          <div
            className="card-container"
>
            
</div>,
        ],
        "className": "search-results",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Search Results",
          },
          "ref": null,
          "rendered": "Search Results",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              Array [],
              null,
              null,
            ],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            null,
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
